### ISSUE TITLE
Addition of Resnet18 Model in Application

### ISSUE THREAD

**Issue by MrinalTyagi** [tensorflow/tensorflow#52231]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.6.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.** Addition of a new pre-trained model: Resnet18 in the application section.

**Will this change the current api? How?** Yes. Addition of a new Resnet18 pre-trained model in the application section.

**Who will benefit with this feature?** Researchers as well as individuals who want to fine-tune or train Resnet18 in Tensorflow without switching to PyTorch

**Any Other info.** This will be a helpful implementation to Tensorflow as currently Resnet18 is present in PyTorch but becomes difficult for individuals to implement who are not familiar with Pytorch and have knowledge of Tensorflow.

**Comment by RishabhGarg108** [comment]
Hey, I would like to work on this issue. I have some experience with open source, but I am a newcomer here and some guidance will be much appreciated.

Where can I find the implementation of other similar models in TensorFlow which I can look for reference?

Thanks!

**Comment by tilakrayal** [comment]
@MrinalTyagi ,
Can you please elaborate about your Feature. Also, please specify the Use Cases for this feature. Thanks!

**Comment by MrinalTyagi** [comment]
Addition of following code to the api:

`tf.keras.applications.ResNet18(
    include_top=True, weights='imagenet', input_tensor=None, 
    input_shape=None, pooling=None, classes=1000,
    classifier_activation='softmax', **kwargs
)`

Use Case: This would allow individuals to fine-tune the ResNet18 model pre-trained on the imagenet dataset to be fine-tuned on their custom dataset. As the implementation of ResNet18 is already available in PyTorch and not in Tensorflow, sometimes it becomes difficult for developers with knowledge of Tensorflow and no prior knowledge of Pytorch to implement the following. Hopefully, this will allow better transfer learning for developers and reduce time in writing the ResNet18 model from scratch, then training it on the imagenet dataset and then fine-tuning it on their custom dataset. Let me know if this resolves the doubt or I need to provide more explanation.

**Comment by MrinalTyagi** [comment]
Sure

**Comment by breadbread1984** [comment]
https://github.com/breadbread1984/resnet18-34  I finished coding and pretraining resnet18. you may try it.
